src/components/AnalysisPage/UnitsSelector/UnitsSelector.jsx

```diff
@@ -1,20 +1,43 @@
 import { useState } from "react";
 import { Select, Space } from "antd";
+import { CheckCircleOutlined } from "@ant-design/icons";
 import "./unitsSelector.css";
 import PropTypes from "prop-types";
 
 const UnitsSelector = ({ partId }) => {
   const [units, setUnits] = useState(undefined);
-  console.log("partId: ", partId);
+  const [showSuccess, setShowSuccess] = useState(false);
 
-  const handleChange = (value) => {
+  const handleChange = async (value) => {
     setUnits(value);
-    console.log(`selected ${value}`);
+    try {
+      const response = await fetch(`/api/v0/parts/${partId}/units`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ units: value }),
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to update units.");
+      }
+
+      const data = await response.json();
+      console.log(data);
```

Comment:
I find HTTP requests to be good things to put into their own functions. Its a lot of setup code that can make the component harder to read through. I'd recommend putting this into its own function and calling it here.

Code for HTTP requests can also tend to grow a lot over time because code for handling different types of error responses might continuously be added. So its good to have it in its own function to put all of that logic and not have it fill up the React component

Reply:
Makes senns